### PR TITLE
Oneil/post /bookings backend validation

### PR DESCRIPTION
**Pull request by Neil0696**
Done the validation task for name, date and if the desks are no longer available.

### CODE REVIEW

File: client/src/pages/ModalBookingScreen.js [hunk 1]
```diff
@@ -129,9 +141,13 @@ function ModalBookingScreen({ bookingDate, refreshBooking, desks }) {
 									checked={dontSelectDesk}
 								/>
 							</Form.Field>
+							{deskErrorMessage && (
+								<p style={{ color: "maroon" }}>{deskErrorMessage}</p>
```
Comment (rebecalsantos) [inline client/src/pages/ModalBookingScreen.js:145]: instead of css styling here you could use a semantic UI <Message>?

File: server/api.js [hunk 2]
```diff
@@ -55,18 +55,37 @@ router.post("/bookings", async function (req, res) {
...
 		}
+
+		const bookingsByDayResult = await db.query(
+			"select * from booking where booking_date = $1",
+			[bookingDate]
+		);
+
+		if (bookingsByDayResult.rows.length >= 5) {
+			return res.status(400).send({
+				message: `No desks available for ${bookingDate}`,
```
Comment (rebecalsantos) [inline server/api.js]: bookingDate needs formating with moment.js (the same as in front end)

File: server/api.js [hunk 3]
```diff
@@ -55,18 +55,37 @@ router.post("/bookings", async function (req, res) {
...
+				field: "date",
+			});
+		}
+
+		if (deskId) {
```
Comment (rebecalsantos) [inline server/api.js]: instead of if(DeskId) maybe put if (deskId !== undefined) maybe deskId = 0 is a real deskId

File: server/api.js [hunk 4]
```diff
@@ -55,18 +55,37 @@ router.post("/bookings", async function (req, res) {
...
+			const test = bookingsByDayResult.rows.find((e) => e.desk_id === deskId);
```
Comment (rebecalsantos) [inline server/api.js]: maybe instead of naming test, isDeskIdAlreadyTaken has got more meaning (is at the beginning because it's a boolean)

**Review by rebecalsantos — CHANGES_REQUESTED** [review]
Thanks for letting me be part of reviewing your code. I really like your consistency in your code and your naming. I have some minor suggestions but everything looks absolutely fine to me. Well done!

**Comment by Neil0696** [comment]
Thank you @rebecalsantos for taking your time to give me some valuable feedback. Great suggestions! I've made the changes now :)

File: server/api.js (hunk 4 above)
Comment (Neil0696) [reply server/api.js]: Great spot! I forgot to change it before doing PR haha

**Review by rebecalsantos — APPROVED** [review]
Thanks for the changes. For me everything is fine.

**Pull request merged**